tests/dip_sip/test_dip_sip.py

```diff
@@ -0,0 +1,168 @@
+
+
+def lag_facts(dut):
+    facts = {}
+    mg_facts = dut.minigraph_facts(host=dut.hostname)['ansible_facts']
+    if not mg_facts['minigraph_portchannels']:
+        pytest.fail("minigraph_portchannels is not defined")
+    host_facts = dut.setup()['ansible_facts']
+    # minigraph facts
+    src_lag = mg_facts['minigraph_portchannel_interfaces'][2]['attachto']
+    dst_lag = mg_facts['minigraph_portchannel_interfaces'][0]['attachto']
+    facts['src_port'] = src_lag
+    facts['dst_port'] = dst_lag
+    logger.info("src_lag is {}, dst_lag is {}".format(src_lag, dst_lag))
+    # lldp facts
+    lldp_facts = dut.lldp()['ansible_facts']['lldp']
+    facts['dst_host_mac'] = lldp_facts[mg_facts['minigraph_portchannels'][dst_lag]['members'][0]]['chassis']['mac']
+    facts['src_host_mac'] = lldp_facts[mg_facts['minigraph_portchannels'][src_lag]['members'][0]]['chassis']['mac']
+    facts['dst_router_mac'] = host_facts['ansible_' + dst_lag]['macaddress']
+    facts['src_router_mac'] = host_facts['ansible_' + src_lag]['macaddress']
+    facts['dst_router_ipv4'] = host_facts['ansible_' + dst_lag]['ipv4']['address']
+    dst_ipv6 = host_facts['ansible_' + dst_lag]['ipv6']
+    facts['dst_router_ipv6'] = [(item['address']) for item in dst_ipv6 if item['scope'] == 'global'][0]
+    facts['dst_port_ids'] = [mg_facts['minigraph_port_indices'][mg_facts['minigraph_portchannels'][dst_lag]['members'][0]]]
+    facts['src_port_ids'] = [mg_facts['minigraph_port_indices'][mg_facts['minigraph_portchannels'][src_lag]['members'][0]]]
+
+    return facts
+
+
+def port_facts(dut):
+    facts = {}
+    mg_facts = dut.minigraph_facts(host=dut.hostname)['ansible_facts']
+
+    if not mg_facts['minigraph_interfaces']:
+        pytest.fail("minigraph_interfaces is not defined.")
+    host_facts = dut.setup()['ansible_facts']
+    # minigraph facts
+    src_port = mg_facts['minigraph_interfaces'][2]['attachto']
+    dst_port = mg_facts['minigraph_interfaces'][0]['attachto']
+    facts['src_port'] = src_port
+    facts['dst_port'] = dst_port
+    logger.info("src_port is {}, dst_port is {}".format(src_port, dst_port))
+    # lldp facts
+    lldp_facts = dut.lldp()['ansible_facts']['lldp']
+    facts['dst_host_mac'] = lldp_facts[dst_port]['chassis']['mac']
+    facts['src_host_mac'] = lldp_facts[src_port]['chassis']['mac']
+    facts['dst_router_mac'] = host_facts['ansible_' + dst_port]['macaddress']
+    facts['src_router_mac'] = host_facts['ansible_' + src_port]['macaddress']
+    facts['dst_router_ipv4'] = host_facts['ansible_' + dst_port]['ipv4']['address']
+    dst_ipv6 = host_facts['ansible_' + dst_port]['ipv6']
+    facts['dst_router_ipv6'] = [(item['address']) for item in dst_ipv6 if item['scope'] == 'global'][0]
+    facts['dst_port_ids'] = [mg_facts['minigraph_port_indices'][dst_port]]
+    facts['src_port_ids'] = [mg_facts['minigraph_port_indices'][src_port]]
+
+    return facts
+
+
+@pytest.fixture(scope='function')
+def gather_facts(testbed_devices, testbed):
```

Comment:
suggest to move to common? @wangxin , @yxieca

Reply:
This one is even more test specific than the above two functions `lag_facts` and `port_facts`.